doc/architecture/architecture.rst

```diff
@@ -1,16 +1,190 @@
 .. _architecture:
 
 Ostro |trade| OS Architecture
-##################################
+#############################
 
-The Ostro OS is a Linux*-based operating system tailored for IoT devices and built with
-security in mind.  Ostro OS is based on standard libraries and services with 
-frameworks to help with best practices and common IoT use cases.
+Introduction
+============
 
-The following documents start off our descriptions of the Ostro |trade| OS 
-components and their relationships so developers
-can learn more technical details about what we're building.
+The Ostro |trade| OS is a pre-compiled, configured and secured base
+Internet of Things (IoT) Linux\* OS that supports creating custom images
+easily. For more information, see :ref:`about_ostro`.
 
+This document gives an overview to Ostro OS architecture and the basic
+building blocks. Furthermore, the document briefly talks about how
+Ostro OS meets different end-user use-cases ranging from hacking to
+IoT product creation.
+
+.. _`Yocto Project`: http://yoctoproject.org
+.. _`Soletta Project`: http://solettaproject.org
+.. _`Node.JS`: https://nodejs.org
+.. _`IoTivity`: https://www.iotivity.org/
+
+Ostro |trade| OS Characteristics
+=================================
+
+In Ostro |trade| OS, there are four key characteristics that
+drive the architecture design:
+
+1. Ostro OS uses standard libraries and services: The building blocks
+used in Ostro OS are commonly used in open source and any Ostro OS
+custom implementation is avoided. This has a big benefit because it
+means there is no extra porting/platform adaptation needed to get
+existing applications or services run in Ostro OS.
+
+2. Ostro OS is pre-compiled and configured: Ostro OS is based
+on the `Yocto Project_` and is provided
+as pre-compiled and configured distribution for IoT. There is no
+end-user compilation needed when starting from scratch with Ostro OS
+since the build system is configured to use the shared. In addition,
+pre-compiled reference images are available for download.
+
+3. Ostro OS is secured and built with security in mind: Security is
+critical for IoT and it is also a focus in Ostro OS. The security
+design is to provide scalable options how security can be configured
+in Ostro based products. The security architecture is explained in
+more details in :ref: `system-and-security-architecture`.
+
+4. Ostro OS is easy to customize: While Ostro OS comes pre-compiled
+(binaries and images), it is important that the OS can be customized
```

Comment:
I would emphasize that the pre-compiled images are *not* to be used in production.